mne/surface.py

```diff
@@ -1689,12 +1691,107 @@ def marching_cubes(image, level):
     mc.Update()
     polydata = mc.GetOutput()
 
+    # get verts and triangles
     verts = numpy_support.vtk_to_numpy(polydata.GetPoints().GetData())
     triangles = numpy_support.vtk_to_numpy(
         polydata.GetPolys().GetConnectivityArray()).reshape(-1, 3)
     verts = np.flip(verts, axis=1)
     triangles = np.flip(triangles, axis=1)
-    return verts, triangles
+
+    # compute curvature and get it
+    curvature = vtkCurvatures()
+    curvature.SetCurvatureTypeToGaussian()
+    curvature.SetInputData(polydata)
+    curvature.Update()
+    curv_data = curvature.GetOutput()
+    curv = numpy_support.vtk_to_numpy(
+        curv_data.GetPointData().GetArray("Gauss_Curvature"))
+    return verts, triangles, curv
+
+
+def make_montage_volume():
+    """Make a volume from montage channel locations.
+
+    Find areas of the input volume with intensity greater than
+    a threshold surrounding local maxima near the channel location.
+    Monotonicity from the peak is enforced to prevent channels bleeding
+    into each other.
+
+    .. note:: This is applicable for channels inside the brain
+              (intracranial electrodes).
+
+    """
+    import nibabel as nib
+    # convert electrode positions from surface RAS to voxels
+    ch_coords = mne.transforms.apply_trans(
+        np.linalg.inv(subject_brain.header.get_vox2ras_tkr()), ch_coords)
+    # take channel coordinates and use the CT to transform them
+    # into a 3D image where all the voxels over a threshold nearby
+    # are labeled with an index
+    CT_data = CT_aligned.get_fdata()
+    thresh = np.quantile(CT_data, CT_thresh)
+    elec_image = np.zeros(subject_brain.shape, dtype=int)
+    for i, ch_coord in enumerate(ch_coords):
+        # this looks up to a voxel away, it may be marked imperfectly
+        volume = mne.voxel_neighbors(ch_coord, CT_data, thresh)
+        for voxel in volume:
+            if elec_image[voxel] != 0:
+                # some voxels ambiguous because the contacts are bridged on
+                # the CT so assign the voxel to the nearest contact location
+                dist_old = np.sqrt(
+                    (ch_coords[elec_image[voxel] - 1] - voxel)**2).sum()
+                dist_new = np.sqrt((ch_coord - voxel)**2).sum()
+                if dist_new < dist_old:
+                    elec_image[voxel] = i + 1
+            else:
+                elec_image[voxel] = i + 1
+
+    # apply the mapping
+    elec_image = nib.Nifti1Image(elec_image, subject_brain.affine)
+
+    # ensure that each electrode contact was marked in at least one voxel
+    assert set(np.arange(1, ch_coords.shape[0] + 1)).difference(
+        set(np.unique(elec_image.get_fdata()))) == set()
```

Comment:
Possibly unrelated cruft since it's not used in the example? Probably makes sense to split into multiple PRs:

1. one that makes it easy to do this electrode mapping
2. another that deals with extracting ROIs associated with electrodes
3. another that deals with visualization of volumetric atlas / aseg in _Brain